boxsdk/object/trash.py

```diff
@@ -0,0 +1,120 @@
+# coding: utf-8
+from __future__ import unicode_literals, absolute_import
+
+import json
+
+from .base_endpoint import BaseEndpoint
+from ..pagination.limit_offset_based_object_collection import LimitOffsetBasedObjectCollection
+
+
+class Trash(BaseEndpoint):
+    """Box API endpoint for performing trash related actions in Box."""
+
+    def get_from_trash(self, item, fields=None):
+        """
+        Get item from trash.
+
+        :param item:
+            The :class:`File` or :class:`Folder` or :class:`WebLink` object to restore from trash.
+        :type item:
+            :class:`File` or :class:`Folder` or :class:`WebLink`.
+        :param fields:
+            List of fields to request
+        :type fields:
+            `Iterable` of `unicode`
+        :returns:
+            A trashed :class:`File`, :class:`Folder`, or :class:`WebLink` object.
+        :rtype:
+            :class:`File`, :class:`Folder`, or :class:`WebLink`
+        """
+        url = self._session.get_url(item.object_type + 's', item.object_id, 'trash')
+        params = {}
+        if fields:
+            params['fields'] = ','.join(fields)
+        box_response = self._session.get(url, params=params)
+        response = box_response.json()
+        return self.translator.translate(response['type'])(
+            session=self._session,
+            object_id=response['id'],
+            response_object=response,
+        )
+
+    def restore_from_trash(self, item, name=None, parent_id=None, fields=None):
```

Comment:
This method should also probably be renamed; it doesn't need to restate that it applies to trash, given that it's a method on `Trash`